tests/unit/l1toL2MessageEvents.test.ts

```diff
@@ -0,0 +1,322 @@
+        },
+        {
+          transactionIndex: 323,
+          blockNumber: 15500657,
+          transactionHash:
+            '0x00000a61331187be51ab9ae792d74f601a5a21fb112f5b9ac5bccb23d4d5aaba',
+          address: '0xd92023E9d9911199a6711321D1277285e6d4e2db',
+          topics: [
+            '0xc1d1490cf25c3b40d600dfb27c7680340ed1ab901b7e8f3551280968a3b372b0',
+            '0x000000000000000000000000a2e06c19ee14255889f0ec0ca37f6d0778d06754',
+            '0x0000000000000000000000006c411ad3e74de3e7bd422b94a27770f5b86c623b',
+            '0x000000000000000000000000000000000000000000000000000000000000504c',
+          ],
+          data: '0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000001442e567b36000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2000000000000000000000000a2e06c19ee14255889f0ec0ca37f6d0778d06754000000000000000000000000a2e06c19ee14255889f0ec0ca37f6d0778d067540000000000000000000000000000000000000000000000000853a0d2313c000000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
+          logIndex: 448,
+          blockHash:
+            '0xe5b6457bc2ec1bb39a88cee7f294ea3ad41b76d1069fd2e69c5959b4ffd6dd56',
+          removed: false,
+        },
+        {
+          transactionIndex: 323,
+          blockNumber: 15500657,
+          transactionHash:
+            '0x00000a61331187be51ab9ae792d74f601a5a21fb112f5b9ac5bccb23d4d5aaba',
+          address: '0xd92023E9d9911199a6711321D1277285e6d4e2db',
+          topics: [
+            '0xb8910b9960c443aac3240b98585384e3a6f109fbf6969e264c3f183d69aba7e1',
+            '0x000000000000000000000000a2e06c19ee14255889f0ec0ca37f6d0778d06754',
+            '0x000000000000000000000000a2e06c19ee14255889f0ec0ca37f6d0778d06754',
+            '0x000000000000000000000000000000000000000000000000000000000000504c',
+          ],
+          data: '0x000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc20000000000000000000000000000000000000000000000000853a0d2313c0000',
+          logIndex: 449,
+          blockHash:
+            '0xe5b6457bc2ec1bb39a88cee7f294ea3ad41b76d1069fd2e69c5959b4ffd6dd56',
+          removed: false,
+        },
+      ],
+      cumulativeGasUsed: constants.Zero,
+      blockNumber: 15500657,
+      confirmations: 1033491,
+      effectiveGasPrice: constants.Zero,
+      byzantium: true,
+      type: 2,
+      status: 1,
+    }
+
+    const arbProvider = new JsonRpcProvider('https://arb1.arbitrum.io/rpc')
+    const l1TxnReceipt = new L1TransactionReceipt(receipt)
+
+    let txReceipt
+    try {
```

Comment:
`.to.throw` doesn't work on Promises. This is the solution I could come up with, I'm open to suggestions.

Reply:
Yeh, i just do it this way too. There some async support for expect, but you need to return it at the end of the func so it doesnt work in lots of cases.